tests/core/pyspec/eth2spec/test/helpers/attestations.py

```diff
@@ -219,35 +219,56 @@ def add_attestations_to_state(spec, state, attestations, slot):
         spec.process_attestation(state, attestation)
 
 
-def next_epoch_with_attestations(spec,
+def next_slots_with_attestations(spec,
                                  state,
+                                 slot_count,
                                  fill_cur_epoch,
-                                 fill_prev_epoch):
-    assert state.slot % spec.SLOTS_PER_EPOCH == 0
-
+                                 fill_prev_epoch,
+                                 participation_fn=None):
+    """
+    participation_fn: (slot, committee_index, committee_indices_set) -> participants_indices_set
+    """
     post_state = state.copy()
     signed_blocks = []
-    for _ in range(spec.SLOTS_PER_EPOCH):
+    for _ in range(slot_count):
         block = build_empty_block_for_next_slot(spec, post_state)
         if fill_cur_epoch and post_state.slot >= spec.MIN_ATTESTATION_INCLUSION_DELAY:
             slot_to_attest = post_state.slot - spec.MIN_ATTESTATION_INCLUSION_DELAY + 1
             committees_per_slot = spec.get_committee_count_per_slot(state, spec.compute_epoch_at_slot(slot_to_attest))
```

Comment:
Seems like we may be able to make the inner `if` for prev and cur an internal helper function, passing in `slot_to_attest` and return an attestation`. Might help reduce the code duplication